astrolabe/runner.py

```diff
@@ -163,34 +167,72 @@ def run(self, persist_cluster=False, startup_time=1):
             driver_workload=self.spec.driverWorkload,
             startup_time=startup_time)
 
-        # Step-3: begin maintenance routine.
-        final_config = self.spec.maintenancePlan.final
-        cluster_config = final_config.clusterConfiguration
-        process_args = final_config.processArgs
-
-        if not cluster_config and not process_args:
-            raise RuntimeError("invalid maintenance plan")
-
-        if cluster_config:
-            LOGGER.info("Pushing cluster configuration update")
-            self.cluster_url.patch(**cluster_config)
-
-        if process_args:
-            LOGGER.info("Pushing process arguments update")
-            self.cluster_url.processArgs.patch(**process_args)
-
-        # Sleep before polling to give Atlas time to update cluster.stateName.
-        sleep(3)
-
-        # Step-4: wait until maintenance completes (cluster is IDLE).
-        selector = BooleanCallablePoller(
-            frequency=self.config.polling_frequency,
-            timeout=self.config.polling_timeout)
-        LOGGER.info("Waiting for cluster maintenance to complete")
-        selector.poll([self], attribute="is_cluster_state", args=("IDLE",),
-                      kwargs={})
-        self.verify_cluster_configuration_matches("final")
-        LOGGER.info("Cluster maintenance complete")
+        for operation in self.spec.operations:
+            if hasattr(operation, 'setClusterConfiguration'):
```

Comment:
Since a single entry in `spec.operations` can only have one operation, I suggest the following refactor to make this all a little easier to read:

```
op_name, op_spec = next(iteritems(operation))

if op_name == 'setClusterConfiguration':
    cluster_config = op_spec.clusterConfiguration
    process_args = op_spec.processArgs
    ...
elif op_name == 'testFailover':
    ...
```

Reply:
Done, thanks for the suggestion.